modules/pam_setquota/pam_setquota.c

```diff
@@ -0,0 +1,238 @@
+/* PAM setquota module
+
+   This PAM module sets disk quota when a session begins.
+
+	 Copyright © 2006 Ruslan Savchenko <savrus@mexmat.net>
+	 Copyright © 2010 Shane Tzen <shane@ict.usc.edu>
+	 Copyright © 2012 Sven Hartge <sven@svenhartge.de>
+   Copyright © 2016 Keller Fuchs <kellerfuchs@hashbang.sh>
+
+*/
+
+#include <sys/types.h>
+#include <sys/quota.h>
+#include <linux/quota.h>
+#include <pwd.h>
+#include <syslog.h>
+#include <errno.h>
+#include <mntent.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#define PAM_SM_SESSION
+
+#include <security/pam_modules.h>
+#include <security/_pam_macros.h>
+#include <security/pam_ext.h>
+#include <security/pam_modutil.h>
+
+#define LOGIN_DEFS "/etc/login.defs"
+
+struct pam_params {
+  uid_t start_uid;
+  uid_t end_uid;
+  char *fs;
+  int overwrite;
+  int debug;
+};
+
+static inline void debug(pam_handle_t *pamh, struct if_dqblk *p) {
+  pam_syslog(pamh, LOG_DEBUG, "bsoftlimit=%llu bhardlimit=%llu"
+                              "isoftlimit=%llu ihardlimit=%llu",
+             p->dqb_bsoftlimit, p->dqb_bhardlimit, p->dqb_isoftlimit,
+             p->dqb_ihardlimit);
+}
+
+static bool _pam_parse_dqblk(int argc, const char **argv, struct if_dqblk *p) {
+  bool bhard = false, bsoft = false, ihard = false, isoft = false;
```

Comment:
Another modern C feature that we do not use anywhere else - bool type. I am not sure we want to introduce it here. Could you please replace with int and true/false with 1/0?

Reply:
https://github.com/linux-pam/linux-pam/pull/94/commits/2b2a3521abdad2ab148ae84ecedef154ca906b7c should fix this.